lib/rules/linebreak-style.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * @fileoverview Rule to forbid mixing LF and LFCR line breaks.
+ * @author Erik Mueller
+ * @copyright 2015 Erik Mueller. All rights reserved.
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function (context) {
+    var linebreakStyle = context.options[0] || "unix";
+
+    return {
+        "Program": function checkForlinebreakStyle(node) {
+            var source = context.getSource(),
+                expectedLF = linebreakStyle === "unix",
+                // Assume LF if we find not one CRLF
+                isSolelyLF = source.indexOf("\r\n") === -1,
+                // Assume CRLF if we find no LF
+                isSolelyCRLF = source.match(/[^\r]\n/) === null;
+
+            if (expectedLF && isSolelyCRLF) {
+                context.report(node, "Expected linebreaks to be 'LF' but found 'CRLF'.");
```

Comment:
We need to find a way to get the line and column for the warning. Right now, this will just report 1:0, so the user won't know where the issue is.